lib/ASTGen/Sources/ASTGen/DeclAttrs.swift

```diff
@@ -55,17 +55,9 @@ extension ASTGenVisitor {
     }
 
     // '@' attributes.
-    // FIXME: Factor out this and share it with 'generate(accessorDecl:)'.
-    var initContext: BridgedPatternBindingInitializer? = nil
-    visitIfConfigElements(node.attributes, of: AttributeSyntax.self) { element in
-      switch element {
-      case .ifConfigDecl(let ifConfigDecl):
-        return .ifConfigDecl(ifConfigDecl)
-      case .attribute(let attribute):
-        return .underlying(attribute)
-      }
-    } body: { attribute in
-      addAttribute(self.generateDeclAttribute(attribute: attribute, initContext: &initContext))
+    var initContext: BridgedPatternBindingInitializer?
+    self.generateDeclAttributes(attributeList: node.attributes, initContext: &initContext) { attr in
+      attrs.add(attr)
```

Comment:
Should this continue to use `addAttribute` to benefit from the presumed future duplicate diagnostic?

Reply:
Let's use `addAttribute` for now :+1:, though I haven't made my mind about the diagnostics. Accessors and closures also need duplicate check. Perhaps `TypeCheckAttr` is the right place?

Reply:
> Perhaps `TypeCheckAttr` is the right place?

Yeah maybe